pifi/videoprocessor.py

```diff
@@ -170,24 +170,31 @@ def __process_and_play_video(self):
             np_array_shape.append(3)
 
         vid_start_time = None
-        frame_length = 1 / fps
+        # Default frame length, until we discover the true one from the fps_fifo.
+        frame_length = 1 / 30.
         last_frame = None
         vid_processing_lag_counter = 0
         is_ffmpeg_done_outputting = False
         avg_color_frames = ReadOnceCircularBuffer(self.__FRAMES_BUFFER_LENGTH)
         ffmpeg_to_python_fifo = open(ffmpeg_to_python_fifo_name, 'rb')
+        fps_fifo = open(fps_fifo_name, 'r')
         while True:
             if is_ffmpeg_done_outputting or avg_color_frames.is_full():
                 pass
             else:
-                is_ffmpeg_done_outputting, vid_start_time = self.__populate_avg_color_frames(
-                    avg_color_frames, ffmpeg_to_python_fifo, vid_start_time, bytes_per_frame, np_array_shape
+                is_ffmpeg_done_outputting, vid_start_time, fps = self.__populate_avg_color_frames(
+                    avg_color_frames, ffmpeg_to_python_fifo, fps_fifo, vid_start_time, bytes_per_frame, np_array_shape
                 )
+                if fps != 0:
+                    frame_length = 1/fps
 
             if vid_start_time is None:
                 # video has not started being processed yet
                 pass
             else:
+                if self.__startup != 0:
+                    self.__logger.info("started playing in {} secs".format(time.time() - self.__startup))
+                    self.__startup = 0
```

Comment:
I don't see where `self.__startup` is initialized?

Reply:
Oops, left it out of this commit.